solver/src/solver/matcha_solver.rs

```diff
@@ -0,0 +1,439 @@
+        let settlement = GPv2Settlement::deployed(&web3).await.unwrap();
+
+        let weth = WETH9::deployed(&web3).await.unwrap();
+        let gno = shared::addr!("6810e776880c02933d47db1b9fc05908e5386b96");
+
+        let solver = MatchaSolver::new(web3, settlement, chain_id).unwrap();
+        let settlement = solver
+            .settle_order(
+                Order {
+                    order_creation: OrderCreation {
+                        sell_token: weth.address(),
+                        buy_token: gno,
+                        sell_amount: 1_000_000_000_000_000_000u128.into(),
+                        buy_amount: 1u128.into(),
+                        kind: OrderKind::Sell,
+                        ..Default::default()
+                    },
+                    ..Default::default()
+                }
+                .into(),
+            )
+            .await
+            .unwrap();
+
+        println!("{:#?}", settlement);
+    }
+
+    #[tokio::test]
+    #[ignore]
+    async fn solve_buy_order_on_matcha() {
+        let web3 = Web3::new(create_env_test_transport());
+        let chain_id = web3.eth().chain_id().await.unwrap().as_u64();
+        let settlement = GPv2Settlement::deployed(&web3).await.unwrap();
+
+        let weth = WETH9::deployed(&web3).await.unwrap();
+        let gno = shared::addr!("6810e776880c02933d47db1b9fc05908e5386b96");
+
+        let solver = MatchaSolver::new(web3, settlement, chain_id).unwrap();
+        let settlement = solver
+            .settle_order(
+                Order {
+                    order_creation: OrderCreation {
+                        sell_token: weth.address(),
+                        buy_token: gno,
+                        sell_amount: 1_000_000_000_000_000_000u128.into(),
+                        buy_amount: 1_000_000_000_000_000_000u128.into(),
+                        kind: OrderKind::Buy,
+                        ..Default::default()
+                    },
+                    ..Default::default()
+                }
+                .into(),
+            )
+            .await
+            .unwrap();
+
+        println!("{:#?}", settlement);
+    }
+
+    #[tokio::test]
+    async fn test_satisfies_limit_price_for_sell_order() {
+        let mut client = Box::new(MockMatchaApi::new());
+        let mut allowance_fetcher = Box::new(MockAllowanceManaging::new());
+
+        let sell_token = H160::from_low_u64_be(1);
+        let buy_token = H160::from_low_u64_be(1);
+
+        let allowance_target = shared::addr!("def1c0ded9bec7f1a1670819833240f027b25eff");
+        client.expect_get_swap().returning(move|_| {
+            Ok(SwapResponse {
+            sell_amount: U256::from_dec_str("100").unwrap(),
+             buy_amount: U256::from_dec_str("91").unwrap(),
+             allowance_target,
+            price: 0.91_f64,
+            to: shared::addr!("def1c0ded9bec7f1a1670819833240f027b25eff"),
+            data: web3::types::Bytes(hex::decode(
```

Comment:
nit: to make this test easier to read, I'd maybe use default or dummy values wherever possible.